File: app/src/main/java/com/kubrik/mex/model/ConnectionState.java
package com.kubrik.mex.model;

public record ConnectionState(
        String connectionId,
        Status status,
        String serverVersion,
        String lastError
) {
    public enum Status { DISCONNECTED, CONNECTING, CONNECTED, ERROR }

    public static ConnectionState disconnected(String id) {
        return new ConnectionState(id, Status.DISCONNECTED, null, null);
    }
}

File: v2/src/shared/connectionState.ts
export type ConnectionState =
  | { kind: 'disconnected' }
  | { kind: 'connecting' }
  | {
      kind: 'connected';
      pingMs: number;
      serverVersion: string;
      topology: string;
      connectedAt: number;
    }
  | { kind: 'error'; message: string };

export interface ConnectionStateChange {
  id: string;
  state: ConnectionState;
}

export interface TestResult {
  ok: boolean;
  pingMs?: number;
  serverVersion?: string;
  topology?: string;
  error?: string;
}
